### ISSUE TITLE
purging ua leaves python dir behind

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-client#326]
On trusty, installing and purging ua leaves this behind:
```
ubuntu@trusty-ua:~$ sudo apt purge ubuntu-advantage-tools 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  ubuntu-advantage-tools*
0 upgraded, 0 newly installed, 1 to remove and 10 not upgraded.
After this operation, 231 kB disk space will be freed.
Do you want to continue? [Y/n] 
(Reading database ... 33198 files and directories currently installed.)
Removing ubuntu-advantage-tools (19.1-0ubuntu1~377~git.bb3519f~ubuntu14.04.1) ...
Purging configuration files for ubuntu-advantage-tools (19.1-0ubuntu1~377~git.bb3519f~ubuntu14.04.1) ...
dpkg: warning: while removing ubuntu-advantage-tools, directory '/usr/lib/python3/dist-packages/uaclient/entitlements' not empty so not removed
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
ubuntu@trusty-ua:~$ l /usr/lib/python3/dist-packages/uaclient/entitlements
total 9.5K
drwxr-xr-x 3 root root  3 Apr  8 21:27 .
drwxr-xr-x 4 root root  4 Apr  8 21:27 ..
drwxr-xr-x 2 root root 10 Apr  8 21:27 __pycache__
```
This may be related to issue #281

**Comment by panlinux** [comment]
Oh, that `__pycache__` dir is full of `pyc` files:
```
-rw-r--r-- 1 root root 6.1K Apr  8 21:27 base.cpython-34.pyc
-rw-r--r-- 1 root root 2.0K Apr  8 21:27 cc.cpython-34.pyc
-rw-r--r-- 1 root root 1.8K Apr  8 21:27 cis.cpython-34.pyc
-rw-r--r-- 1 root root 1.8K Apr  8 21:27 esm.cpython-34.pyc
-rw-r--r-- 1 root root 3.0K Apr  8 21:27 fips.cpython-34.pyc
-rw-r--r-- 1 root root  773 Apr  8 21:27 __init__.cpython-34.pyc
-rw-r--r-- 1 root root 5.5K Apr  8 21:27 livepatch.cpython-34.pyc
-rw-r--r-- 1 root root 4.0K Apr  8 21:27 repo.cpython-34.pyc
```

**Comment by panlinux** [comment]
IMPORTANT: this was based on bb3519fd7491bc24e2b0f1ab4dfe1f3493ee99ab, latest build in the daily ppa. Let's look again after the latest builds.

**Comment by powersj** [comment]
On my xenial container with the latest PPA build from tonight `ubuntu-advantage-tools - 19.2~392~git.e0ecc40~ubuntu16.04.1`, 1) shows those same files 2) on removal of the package the directory is gone.

**Comment by powersj** [comment]
During the build it shows the creation of these files:

byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/defaults.py to defaults.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/sso.py to sso.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/__init__.py to __init__.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/version.py to version.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/apt.py to apt.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/base.py to base.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/cis.py to cis.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/cc.py to cc.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/__init__.py to __init__.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/livepatch.py to livepatch.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/esm.py to esm.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/repo.py to repo.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/entitlements/fips.py to fips.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/status.py to status.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/config.py to config.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/cli.py to cli.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/serviceclient.py to serviceclient.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/util.py to util.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/contract.py to contract.cpython-35.pyc
byte-compiling /<<PKGBUILDDIR>>/debian/ubuntu-advantage-tools/usr/lib/python3.5/dist-packages/uaclient/conftest.py to conftest.cpython-35.pyc

**Comment by powersj** [comment]
This appears fixed with a91a1974b38315b146d0ef329a06e131a701b68b
